src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,231 @@
+const REQUEST_TIMEOUT = 8000; // ms
+const request = require('request-promise-native');
+const { Configuration } = require('@schul-cloud/commons')
+
+const DOMAIN = Configuration.get('EDUSHARING__DOMAIN');
+const USER = Configuration.get('EDUSHARING__USER');
+const PASSWORD = Configuration.get('EDUSHARING__PASSWORD');
+const GRANT_TYPE = Configuration.get('EDUSHARING__GRANT_TYPE');
+const OAUTH_SECRET = Configuration.get('EDUSHARING__OAUTH_SECRET');
+const CLIENT_ID = Configuration.get('EDUSHARING__CLIENT_ID');
+
+
+// STACKOVERFLOW BEAUTY
+function validURL(str) {
+	const pattern = new RegExp(
+		'^(https?:\\/\\/)?'
+		+ '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'
+		+ '((\\d{1,3}\\.){3}\\d{1,3}))'
+		+ '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'
+		+ '(\\?[;&a-z\\d%_.~+=-]*)?'
+		+ '(\\#[-a-z\\d_]*)?$',
+		'i'
+	);
+	return !!pattern.test(str);
+}
+
+class EduSharingConnector {
+	constructor() {
+		if (EduSharingConnector.instance) {
+			return EduSharingConnector.instance;
+		}
+		if (!validURL(this.url)) {
+			return 'Invalid domain, check your .env';
+		}
+		this.authorization = null; // JSESSION COOKIE
+		this.accessToken = null; // ACCESSTOKEN
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json'
+		};
+	}
+
+	static get authorization() {
+		const headers = {
+			...EduSharingConnector.headers,
+			Authorization: `Basic ${Buffer.from(`${USER}:${PASSWORD}`).toString(
+				'base64'
+			)}`
+		};
+
+		return headers;
+	}
+
+	// gets cookie (JSESSION) and attach it to header
+	getCookie() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Block must not be padded by blank lines.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4991925)